syntax = "proto3";

package post_service;

service PostService {
    rpc CreatePost(CreatePostRequest) returns (CreatePostResponse);
    rpc GetAllPost(GetAllPostRequest) returns (GetAllPostResponse);
    rpc GetUniquePost(GetUniquePostRequest) returns (GetUniquePostResponse);
    rpc LikePost(LikePostRequest) returns (LikePostResponse);
    rpc CommentPost(CommentPostRequest) returns (CommentPostResponse);
}

message CreatePostRequest {
    int32 user_id = 1;
    string title = 2;
    bytes post_image = 3;
    string content = 4;
    string link = 5;
}


message CreatePostResponse {
    int32 post_id = 1;
    string message = 2;
}


message GetAllPostRequest{
    int32 user_id = 1;
}


message Post {
    int32 post_id = 1;
    int32 user_id = 2;
    string title = 3;
    string content = 4;
    string link = 5;
    string date = 6;
    string postimage = 7;
    bool like = 8;
}

message GetAllPostResponse{
    repeated Post posts = 1;
}


message Comment{
    int32 comment_id =1;
    int32 user_id = 2;
    string content = 3;
    string date = 4;
}


message GetUniquePostRequest{
    int32 post_id = 1;
    int32 user_id = 2;
}



message GetUniquePostResponse{
    int32 post_id = 1;
    int32 user_id = 2;
    string title = 3;
    string content = 4;
    string link = 5;
    string date = 6;
    string postimage = 7;
    bool like = 8;
    repeated Comment comments = 9;
}


message LikePostRequest{
    int32 post_id = 1;
    int32 user_id = 2;
}


message LikePostResponse{
    string message = 1;
}


message CommentPostRequest{
    int32 post_id = 1;
    int32 user_id = 2;
    string content = 3 ;
}


message CommentPostResponse{
    string message = 1;
}




